crates/core/src/node/eth.rs

```diff
@@ -69,7 +74,9 @@ impl InMemoryNode {
                     message
                 );
 
-                sh_eprintln!("\n{}", pretty_message);
+                let halt_error: HaltError = reason.clone().to_halt_error().await;
```

Comment:
Same -- maybe we can pass it up the execution stack and handle in a single location